src/core/server/execution_context/execution_context_service.mock.ts

```diff
@@ -31,6 +31,7 @@ const createInternalSetupContractMock = () => {
 const createSetupContractMock = () => {
   const mock: jest.Mocked<ExecutionContextSetup> = {
     withContext: jest.fn(),
+    setRequestId: jest.fn(),
```

Comment:
This new `setRequestId` fn doesn't appear to be used in the alerting / task manager parts of this PR.  Did you want that included, or does it need to be included for some reason.  Wondering if you accidentally left it in ...

Reply:
I added it to register a request id when Alerting plugin creates a `FakeRequest` instance. See the PR title for instructions (copied):

and sends it to ES server as `x-opaque-id` header:
```
2021-08-11T16:50:13.767+03:00|{"http":{"request":{"id":"undefined;kibana:task manager:run alerting:monitoring_shard_size:a2564810-fa85-11eb-9f0e-e1e1b261f0a9;alert:execute monitoring_shard_size:a1426710-fa85-11eb-9f0e-e1e1b261f0a9"}}}
```
The first `undefined` here is the lack of `x-opaque-id` for a request crated by `alerting`. I added a commit exposing `setRequestId`, feel free to call when creating a fake request
```js
const fakeRequest = ...;
const id = uuid.v4()
executionContext.setRequestId(id);
```

Reply:
Ok, I added a logic not to include requested if not present. We can enhance `FakeRequest` logic to include a unique identifier later. Right now the problem is that `FakeRequest` factory may be called several times during a single task execution, so we either have to pass `id` from outside or make `id` generation idempotent (based on ApiKey, for example).